completion/services.py

```diff
@@ -76,6 +76,44 @@ def fill_in_run(block_key):
                 completions[candidate] = 0.0
         return completions
 
+    def get_selected_library_children(self, library_node):
+        """
+        Content Libraries will select a subset of the content in their library to
+        show to each learner. This function takes in a library content node and
+        returns the subset of its children that were selected for this user.
+        """
+        selected_children = []
+        all_children = library_node.get_children()
+        for block_type, block_id in library_node.selected:
+            selected_key = library_node.location.course_key.make_usage_key(block_type, block_id)
+            selected_children.extend(
+                list(filter(
```

Comment:
Not: I tend to prefer list comprehensions for this sort of thing in python, because python lambdas are a bit wordy.

Less of a nit: `get_children` doesn potentially expensive work for all children, and then you filter them out. You could instwad iterate over `selected` and then load them from the runtime yourself.

Reply:
Switched to list comprehension. Definitely cleaner.

Yeah, I wasn't sure about that for a  couple reasons:
1) The block transformer uses this similar format so I wasn't sure if it was a defensive measure in the situation of a selected block having been removed from the library and thus won't be in the children.
2) I'm not sure how to load from the runtime (modulestore?) (in general) and especially while in the completion service.

Let me know what you think.

Reply:
I think https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/library_content_module.py#L303-L309 could do what you want.

Reply:
Ah yes! There was a method in that file called `get_child_descriptors`that returns the list version of the `_get_selected_child_blocks`. Great call there, Cale!